src/js/randomizer/history.ts

```diff
@@ -87,8 +87,8 @@ export type SetHistory = Dispatch<
 >;
 
 export function pureHistoryApply<T>(
-  setHistory: Dispatch<SetStateAction<History<T>>>,
-  a: PureHistoryAction<T>,
+  setHistory: Dispatch<SetStateAction<History<Partial<T>>>>,
+  a: PureHistoryAction<Partial<T>>,
```

Comment:
Could also remove the generic and make it

```ts
export function pureHistoryApply(
  setHistory: Dispatch<SetHistory>,
  a: PureHistoryAction<Partial<ScriptState>>,
) { /* ... */ }
```

Reply:
Let's go with that. In principle having it generic is nice but there's only type of history in this project so let's just have it reflect that. I'd prefer having it be monomorphic than generic-but-not-as-generic-as-possible.